src/prefect/indicators/extract/e4.py

```diff
@@ -0,0 +1,194 @@
+"""QualiCharge prefect indicators: extract.
+
+E4: the list of points of charge in activity.
+"""
+
+from datetime import datetime
+from string import Template
+from typing import List
+from uuid import UUID
+
+import numpy as np
+import pandas as pd  # type: ignore
+from prefect import flow, runtime, task
+from prefect.cache_policies import NONE
+from prefect.futures import wait
+from prefect.task_runners import ThreadPoolTaskRunner
+from sqlalchemy.orm import Session
+
+from indicators.conf import settings
+from indicators.db import get_api_db_engine
+from indicators.models import IndicatorPeriod, IndicatorTimeSpan, Level, PeriodDuration
+from indicators.types import Environment
+from indicators.utils import (
+    export_indicators,
+    get_num_for_level_query_params,
+    get_targets_for_level,
+    get_timespan_filter_query_params,
+    set_start,
+)
+
+LIST_POCS_FOR_LEVEL_QUERY_TEMPLATE = """
+SELECT
+    id_pdc_itinerance,
+    $level_id AS level_id
+FROM
+    SESSION
+    INNER JOIN PointDeCharge ON point_de_charge_id = PointDeCharge.id
+    LEFT JOIN station ON station_id = station.id
+    LEFT JOIN localisation ON localisation_id = localisation.id
+    LEFT JOIN city ON city.code = code_insee_commune
+    $join_extras
+WHERE
+    $level_id IN ($indexes)
+    AND $timespan
+GROUP BY
+    id_pdc_itinerance,
+    $level_id
+"""
+QUERY_NATIONAL_TEMPLATE = """
+SELECT
+    id_pdc_itinerance
+FROM
+    SESSION
+    INNER JOIN PointDeCharge ON point_de_charge_id = PointDeCharge.id
+WHERE
+    $timespan
+GROUP BY
+    id_pdc_itinerance
+"""
+
```

Comment:
As discussed, we could expect an indicator to define such global:

```suggestion
HISTORY_STRATEGY_FIELD: str = "mean"
```

Reply:
"up" indicator need to know the STRATEGY for each indicator

Reply:
done